bluepyopt/ephys/create_acc.py

```diff
@@ -0,0 +1,348 @@
+'''create a hoc file from a set of BluePyOpt.ephys parameters'''
+
+# pylint: disable=R0914
+
+import os
+
+from collections import namedtuple
+from glob import glob
+
+import numpy
+import jinja2
+
+from .create_hoc import Location, Range, _get_template_params
+
+
+# Define Neuron to Arbor variable conversions
+ArbVar = namedtuple('ArbVar', 'name, conv', 
+                    defaults=[None,None])
+
+_nrn2arb_var = dict(
+    cm=ArbVar(name='membrane-capacitance'), # conv=None implies identity
+    ena=ArbVar(name='ion-reversal-potential \"na\"'),
+    ek=ArbVar(name='ion-reversal-potential \"k\"'),
+    v_init=ArbVar(name='membrane-potential'),
+    celsius=ArbVar(name='temperature-kelvin', 
+                   conv=lambda celsius: celsius + 273.15),
+    Ra=ArbVar(name='axial-resistivity')
+)
+
+
+def _nrn2arb_var_name(name):
+    """Neuron to Arbor variable renaming."""
+    return _nrn2arb_var[name].name if name in _nrn2arb_var else name
+
+
+def _nrn2arb_var_value(param):
+    """Neuron to Arbor variable value conversion."""
+    if param.name in _nrn2arb_var and _nrn2arb_var[param.name].conv is not None:
+        return _nrn2arb_var[param.name].conv(param.value)
+    else:
+        return param.value
+
+
+def _arb_is_global_param(loc, param):
+    """Returns if location-specific variable is a global one in Arbor."""
+    return loc == 'all' and param.name in ['membrane-capacitance']
+
+
+# Define BluePyOpt to Arbor region mapping (relabeling locations to SWC convention)
+# Remarks:
+#  - using SWC convetion: 'dend' for basal dendrite, 'apic' for apical dendrite
+#  - myelinated is unsupported in Arbor
```

Comment:
Can we add support for myelinated in Arbor? Do you know how these section are defined?

Reply:
I'm not completely sure where this is used (it doesn't seem to be in `l5pc`). A quick search finds it in several `.hoc` files in `examples/stochkv`. Does somebody else have insights on this?

Reply:
'myelinated' is a sectionlist we're using in our latest models. It's a cylinder attached to the end of the AIS to prevent boundary effects. It doesn't have channels but just represent a myelinated chunk of axon

Reply:
Thanks @wvangeit for the explanation. I assume AIS means axon initial segment. Is there an example/morphology with `myelinated`? Otherwise, I'd suggest to throw an exception for now until we can properly test that in Arbor.

Reply:
We don't have this in open source models yet. But it's fairly simple code though. It's not specified in the morphology file itself, but used in the replace_axon function when constructing the model:
`
       sim.neuron.h.execute('create myelin[1]', icell)                                                                                                                                                                                 
        icell.myelinated.append(sec=icell.myelin[0])                                                                                                                                                                                    
        icell.all.append(sec=icell.myelin[0])                                                                                                                                                                                           
        icell.myelin[0].nseg = 5                                                                                                                                                                                                        
        icell.myelin[0].L = 1000                                                                                                                                                                                                        
        icell.myelin[0].diam = diams[count-1]                                                                                                                                                                                           
        icell.myelin[0].connect(icell.axon[1], 1.0, 0.0)         
`

Reply:
Ok, what does `count`, `diams[count-1]` respectively evaluate to? And is it correct to assume that `myelinated` can only appear in conjunction with axon replacement?

Reply:
It's just the last diameter of the axon up until that point. 
In this particular case, yes, there will only by myelin after axon replacement. But of course it's up to the user, i could also see some people handling part of the real experimental morphology as myelinated and adapt the parameters accordingly

Reply:
Ok, thanks. When I naively access `self.icell.myelin[0]` in an instantiated cell model, I'm getting a crash 
```
    NEURON: section in the object was deleted
        near line 0
        {create axon[2]}
```
even though `len(self.icell.myelin) == 1` (for e.g. the simple cell morphology, where the above code wasn't run, just the usual `replace_axon`). How can I safely find out that `self.icell.myelin` is not there despite len > 0?

Otherwise, I can add your code above to the `replace_axon` function in the morphology to test the support of this section.

Reply:
Tested - it works with your code above added to `replace_axon`. I've added the necessary changes to c0af192da6, but they're currently commented out as I have no way to figure out if the `myelin` section exists without crashing in Neuron.

Reply:
not sure, but i wonder if hasattr():
https://www.w3schools.com/python/ref_func_hasattr.asp
would work on icell?

Reply:
That works on myelin and the `len(self.icell.myelin) == 1` looks like a bug given that NEURON crashes, complaining that the section was deleted upon accessing e.g. `self.icell.myelin[0]`. I've added safe support for `myelin` now in 2c67258.

Reply:
Let's look at this in a separate issue.

Reply:
See #419.